lib/Test/Net/RabbitMQ.pm

```diff
@@ -156,17 +158,23 @@ has queues => (
 );
 
 has delivery_tag => (
-    traits  => [ qw(Counter) ],
-    is      => 'ro',
-    isa     => 'Num',
-    default => 0,
-    handles => {
-        _inc_delivery_tag   => 'inc',
-        _dec_delivery_tag   => 'dec',
-        _reset_delivery_tag => 'reset',
-    },
+    is      => 'rw',
+    isa     => 'Math::UInt64',
+    default => sub { uint64(0) },
+    clearer => '_reset_delivery_tag',
+    writer  => '_set_delivery_tag',
```

Comment:
You were right to have a writer. I had missed the need for it elsewhere in the code. I ended up merging this as-is.